protocols/identify/src/behaviour.rs

```diff
@@ -184,20 +180,19 @@ impl Config {
         self
     }
 
-    /// Configures the size of the LRU cache, caching addresses of discovered peers.
-    pub fn with_cache_size(mut self, cache_size: usize) -> Self {
-        self.cache_size = cache_size;
+    /// Configuration for the LRU cache responsible for caching addresses of discovered peers.
+    ///
+    /// If set to [`None`], caching is disabled.
+    pub fn with_cache_config(mut self, cache_config: Option<PeerAddressesConfig>) -> Self {
+        self.cache_config = cache_config;
```

Comment:
We should probably keep `with_cache_size`, deprecating it for `with_cache_config` and just have use that internally so we could probably introduce this as a patch update. Thoughts?

Reply:
If this is possible, I'm definitely doing that. I did not think of that thanks.

Reply:
It's done. I'll let you resolve this if you're ok with it. :wink:

Reply:
Ok. It was slightly more complicated than expected because since every attributes of `identify::Config` are public, I cannot remove `cache_size` if we want to stay in `0.45.1` and not upgrade to `0.46.0`.

Just a thought, but I checked the other "`Config`" object across the project and some do expose the attributes but most of them don't. Maybe we should homogenize that and make every attributes private to not have this kind of problem later on ? It would be a pain to make many new major updates but it would once and for all. If you think there is something here, I will happily open an issue to discuss this mater more.

Reply:
@dariusc93 I'm closing this conversation. Feel free to re-open it if you have other suggestions about it.

Reply:
I agree with François on this one, can we just introduce a breaking change, and take the minor release cycle to also do https://github.com/libp2p/rust-libp2p/issues/5660?

Reply:
As discussed with you @jxs, I have removed the deprecations and took advantage of the new `0.46` version introduced.